### PR TITLE
Issue #78 - Shared Pointer Usage Adjustments in Planners

### PR DESCRIPTION
**Pull request by Jmeyer1292**
This pull requests changes the path planner interface to take its robot model shared pointer by value instead of by reference. Other changes to implementations to facilitate this.

### CODE REVIEW

**Comment by jrgnicho** [comment]
Sorry for not replying to the issue sooner, but see my comment [here](https://github.com/ros-industrial-consortium/descartes/issues/78)

File: descartes_planner/src/dense_planner.cpp [hunk 1]
```diff
@@ -31,10 +31,10 @@ DensePlanner::~DensePlanner()
 
 }
 
-bool DensePlanner::initialize(descartes_core::RobotModelConstPtr &model)
+bool DensePlanner::initialize(descartes_core::RobotModelConstPtr model)
 {
   planning_graph_ = boost::shared_ptr<descartes_planner::PlanningGraph>(
-      new descartes_planner::PlanningGraph(model));
+      new descartes_planner::PlanningGraph(std::move(model)));
```
Comment (jrgnicho) [inline descartes_planner/src/dense_planner.cpp:37]: Is this really necessary for passing a shared pointer?

Comment (Jmeyer1292) [inline descartes_planner/src/dense_planner.cpp:37]: Moving the shared pointer avoids an atomic reference access and write. It also shows that we're just passing that value along. It's a small pro, but I think it is 'most' correct. Omitting the move wouldn't hurt the logic of the code.

Comment (jrgnicho) [inline descartes_planner/src/dense_planner.cpp:37]: Would this not prevent incrementing the reference count of the shared pointer? As far as I understand the use of move semantics in this case transfers ownership of the object, so if the containing object goes out of scope could that invalidate other shared pointers?

Comment (Jmeyer1292) [inline descartes_planner/src/dense_planner.cpp:37]: The shared pointer is "anchored" in main. That pointer is copied into the initialize method and then moved down to planning graph. When initialize is done the reference count on the pointer should be 2.

Comment (jrgnicho) [inline descartes_planner/src/dense_planner.cpp:37]: +1

**Pull request merged**